pip/req/req_install.py

```diff
@@ -778,6 +779,14 @@ def install(self, install_options, global_options=(), root=None):
             self.install_succeeded = True
             return
 
+        # Extend the list of global and install options passed on to
+        # the setup.py call with the ones from the requirements file.
+        # Options specified in requirements file override those
+        # specified on the command line, since the last option given
+        # to setup.py is the one that is used.
```

Comment:
I don't think we can easily trim duplicates. For a start you'd need to know that ```-c``` and ```--compile``` are aliases, and that ```--no-compile``` is another variant. And projects could add extra options that might be additive.

I think documenting clearly that install and global options from the requirements file and the command line are both added to the final command, with a clear specification of the order they are added in, plus a note saying that the user is responsible for ensuring that the resulting command is valid, is all we can do.

For example, I use the following ```pip wheel``` command to build pyyaml:

```
pip wheel --global-option "--with-libyaml" --global-option "build_ext" --global-option "-DYAML_DECLARE_STATIC" --global-option "-LC:\Work\Scratch\pyyaml\yaml-0.1.5\win32\vs2008\Output\Release\lib" --global-option "-IC:\Work\Scratch\pyyaml\yaml-0.1.5\include"  .\PyYAML-3.10.tar.gz
```

Note that I'm using a global option of ```build_ext``` to insert an explicit extension build command into the setup.py invocation, because I need the include file options added to the ```build_ext``` command, and the ```bdist_wheel``` command doesn't support them. Putting that lot in a requirements file would be an obvious thing to do.

(I'm not saying this is good practice, it's clearly an abuse, but sometimes you do what you must :-))